File: src/main/java/net/adventuregame/resources/ResourceKey.java
package net.adventuregame.resources;

public class ResourceKey<T> {
    private final ResourceLocation location;

    public ResourceKey(ResourceLocation location) {
        this.location = location;
    }

    public static <T> ResourceKey<T> create(String key) {
        return new ResourceKey<>(ResourceLocation.withDefaultNamespace(key));
    }

    public ResourceLocation location() {
        return location;
    }

    @Override
    public String toString() {
        return location.toString();
    }
}

File: src/main/kotlin/net/adventuregame/resources/ResourceKey.kt
package net.adventuregame.resources

class ResourceKey<T>(private val location: ResourceLocation) {
    fun location(): ResourceLocation {
        return location
    }

    override fun toString(): String {
        return location.toString()
    }

    companion object {
        fun <T> create(key: String?): ResourceKey<T?> {
            return ResourceKey<T?>(ResourceLocation.Companion.withDefaultNamespace(key))
        }
    }
}

